### DISCUSSION TITLE
Disable admin page activates registration for everyone

### DISCUSSION THREAD

**Discussion by CamelT0E** [dani-garcia/vaultwarden#4525] [General]
I would like to disable the admin page. On the Synology you can not set the reverse proxy well, that it is blocked externally.

If the admin page is [deactivated](https://github.com/dani-garcia/vaultwarden/wiki/Hardening-Guide#disable-registration-and-optionally-invitations) (Environments or *.json) this works, but new users can [register](https://github.com/dani-garcia/vaultwarden/wiki/Disable-registration-of-new-users) again. The option that no new users are allowed to register is ignored! After activating the admin page, the new registration block works again!

It would be enough for me to activate the admin interface only if necessary. I didn't find anything about this in the wiki.

**Comment by BlackDex** [comment]
It can be password protected, so i would suggest to do that.
Also, the admin interface is also brute-force protected, you can even make that more tight if you want so that any wrong attempt is blocked for a long time.

https://github.com/dani-garcia/vaultwarden/blob/2ad33ec97f415edb2af054f527efed52b3b93a9e/.env.template#L384-L387

**Reply by CamelT0E** [reply]
maybe "ADMIN_RATELIMIT" would be interesting in the extended admin pages. "Admin session lifetime" is already available.

**Reply by CamelT0E** [reply]
Just out of curiosity: is it possible to unlock ADMIN_RATELIMIT manually? Or is this not provided for?

**Comment by dani-garcia** [comment]
I'm not sure I understand the question, both disabling the admin page and user registrations at the same time is supported, just don't set ADMIN_TOKEN and set SIGNUPS_ALLOWED=false. Wether the admin page is enabled or not is irrelevant to the signups_allowed setting.

 Just tested it in my setup and it works just fine.

**Reply by CamelT0E** [reply]
Admin Page end User registrations at the same time does not work for me. the settings are set in the *.json, so no environments are set in the container. i noticed it only by chance and was the first to look for the error with me. I just disabled the admin token, which also works. After that I can register a new user. If I undo the deactivation, the block will function again.

**Reply by BlackDex** [reply]
How do you _disable_ the admin token in the json file?

You should remove that line fully if you want it to be unset.
If it is unset you won't be able to access it.

Also, those settings have no connection, so they are fully independent.

**Reply by CamelT0E** [reply]
I commented out the line with a # sign. This also worked so far that the admin web interface was deactivated: I am currently testing the complete removal. If this works, is there no better way to disable the admin panel quickly? Unfortunately, it is very difficult to use another reverse proxy with synology.

**Reply by CamelT0E** [reply]
I tried it again. Registration is no longer possible after complete removal. The admin panel gets the same error message as when the token is disabled with #.  Due to this error message, I assumed that the admin page was properly disabled. According to the web interface, it is disabled.

![grafik](https://github.com/dani-garcia/vaultwarden/assets/59985723/1574ff97-1be5-4617-b5af-761563dfa25e)

tested with a clean test installation. the error message is confusing.

**Reply by BlackDex** [reply]
How is that message confusing? It tells you exactly what it is missing.
If there is an error in the config or loading of a file, that isn't visible here.

**Reply by tessus** [reply]
> I commented out the line with a # sign.

There are no comments in JSON, thus adding a # to a line in a json document renders that json file invalid.

**Reply by CamelT0E** [reply]
@tessus, 
I think you're right.
The problem is solved for me with this.
Thanks to all the helpers for the quick help!

**Reply by BlackDex** [reply]
Currently it seems to fail silently, so that is probably not something we want.
What it does then, is it falls back to the env variables.

**Comment by CamelT0E** [comment]
.env does not work in vaultwarden docker?
With this, you could quickly control some environments that are not available in the admin panel.
This would make it possible to activate/deactivate parts without editing the container. A reboot should be enough.

**Reply by BlackDex** [reply]
Restarting Vaultwarden or the container should be enough.

**Reply by CamelT0E** [reply]
.but env is apparently not being read. I've added the ones for a test, the variables for the admin token and web vault. The settings remained without function, and there was nothing to be found in the log either. Which .env was located in the root directory at the *.json & database. Of course, with restart and paused containers.

**Reply by BlackDex** [reply]
.env needs to be right besides the Vaultwarden binary, or you need to set the correct env variable to point it to an other file. 

Without those it will know where to look.

**Comment by CamelT0E** [comment]
The variables with the .env works.
I have added the following environments in the container. `-e ENV_FILE=/data/.env \`
With this, I can make the rest of the [settings](https://github.com/dani-garcia/vaultwarden/blob/main/.env.template).

Problem solved. Many Thanks 👍
